llvm/lib/Transforms/Utils/DemoteRegToStack.cpp

```diff
@@ -102,9 +102,12 @@ AllocaInst *llvm::DemoteRegToStack(Instruction &I, bool VolatileLoads,
         new StoreInst(&I, Slot, Handler->getFirstInsertionPt());
       return Slot;
     }
+  } else if (InvokeInst *II = dyn_cast<InvokeInst>(&I)) {
+    InsertPt = II->getNormalDest()->getFirstInsertionPt();
+  } else if (CallBrInst *CBI = dyn_cast<CallBrInst>(&I)) {
+    InsertPt = CBI->getDefaultDest()->getFirstInsertionPt();
```

Comment:
I don't think this is right. The result of callbr can be used on all destinations, not just the default destination.

Reply:
Why? As the langref says:
> The output values of a ‘callbr’ instruction are available only to the ‘fallthrough’ block, not to any ‘indirect’ blocks(s).

Reply:
This used to be the case, but the limitation no longer exists. Looks like we forgot to drop this note in LangRef.

Reply:
Done now. Do we need to drop the note?

Reply:
Done in https://github.com/llvm/llvm-project/pull/91167.